### PR TITLE
Next Preview Plugin

### PR DESCRIPTION
**Pull request by roshanjrajan-zip**
Deploy from Snapshot page for Next-Preview

image

### CODE REVIEW

File: zip-deployboard-deck/src/snapshots/snapshot.config.ts [hunk 1]
```diff
@@ -25,8 +27,8 @@ export const SNAPSHOT_CONFIGS: Map<string, SnapshotConfigType> = new Map<string,
     'test-build-branch',
     {
       gitBranch: 'test-build-branch',
-      label: 'Next',
-      pipeline: 'Next',
+      label: NEXT_PREVIEW,
```
Comment (zhenchristopher-zip) [inline zip-deployboard-deck/src/snapshots/snapshot.config.ts:30]: should this be all caps?

**Review by zhenchristopher-zip — APPROVED** [review]
nice! I'm gonna trust that this runs locally, lmk if you have issues deploying!

File: zip-deployboard-deck/src/snapshots/snapshot.config.ts (hunk 1 above)
Comment (zhenchristopher-zip) [reply zip-deployboard-deck/src/snapshots/snapshot.config.ts:30]: oh nvm i didn't see it's an enum 🤦

File: zip-deployboard-deck/src/snapshots/Snapshots.tsx [hunk 2]
```diff
@@ -178,12 +191,58 @@ const SnapshotDeployModal = ({ pipeline, setPipeline, image, onClose, onSubmit }
...
+              <Form>
+                <Form.Group className="mb-3" controlId="formBasicEmail">
+                  <Form.Label>Name</Form.Label>
+                  <Form.Control
+                    style={{ maxWidth: '200px' }}
+                    type="name"
+                    placeholder="Enter name"
+                    onChange={(e) => setName(e.target.value.trim())}
+                  />
+                  <Form.Text className="text-muted">You can access this at {name}.next-preview.ziphq.com</Form.Text>
```
Comment (zhenchristopher-zip) [inline zip-deployboard-deck/src/snapshots/Snapshots.tsx]: nit: would be nice if we enclose the link in a `<a href="{name}.next-preview.ziphq.com"></a>` tag so the user can just clink the link

File: zip-deployboard-deck/src/snapshots/Snapshots.tsx [hunk 3]
```diff
@@ -178,12 +191,58 @@ const SnapshotDeployModal = ({ pipeline, setPipeline, image, onClose, onSubmit }
...
+              <label>Select number of hours </label>
+              <Dropdown>
+                <Dropdown.Toggle>
+                  <span style={{ display: 'flex', alignItems: 'center' }}>
+                    {`Hours: ${hours}`}
+                    <ArrowDropDown />
+                  </span>
+                </Dropdown.Toggle>
+                <Dropdown.Menu>
+                  {Array.from([1, 2, 3]).map((hrs) => (
```
Comment (zhenchristopher-zip) [inline zip-deployboard-deck/src/snapshots/Snapshots.tsx]: maybe 0.5, 2, 4?

File: zip-deployboard-deck/src/snapshots/Snapshots.tsx (hunk 2 above)
Comment (roshanjrajan-zip) [reply zip-deployboard-deck/src/snapshots/Snapshots.tsx]: Thats a good idea. Adding!

File: zip-deployboard-deck/src/snapshots/Snapshots.tsx (hunk 3 above)
Comment (roshanjrajan-zip) [reply zip-deployboard-deck/src/snapshots/Snapshots.tsx]: I don't know if anyone would ever choose the 0.5 options, I could do 1 - 4 if that makes more sense?

Comment (zhenchristopher-zip) [reply zip-deployboard-deck/src/snapshots/Snapshots.tsx]: actually yeah, looking through slack it looks like people usually just take it for an hour at least, so this lgtm!

**Pull request merged**